src/networking.c

```diff
@@ -1518,7 +1518,11 @@ void unlinkClient(client *c) {
             }
         }
         /* Only use shutdown when the fork is active and we are the parent. */
-        if (server.child_type) connShutdown(c->conn);
+        if (server.child_type && !c->flag.repl_rdb_channel) {
+            connShutdown(c->conn);
+        } else if (c->flag.repl_rdb_channel) {
+            shutdown(c->conn->fd, SHUT_RDWR);
```

Comment:
This pierces the connection abstraction, which we don't want to do. Connections are supposed to be opaque to us. This is also documented clearly enough to understand why we are shutting down the repl channel in the TLS case. Maybe add a flag to connShutdown here.